lib/src/message/message_item_bloc.dart

```diff
@@ -171,49 +245,50 @@ class MessageItemBloc extends Bloc<MessageItemEvent, MessageItemState> {
     ]);
   }
 
-  Future<void> _setupNextMessage() async {
-    await _getNextMessage().loadValue(ChatMsg.methodMessageGetTimestamp);
+  Future<void> _setupNextMessage(int nextMessageId) async {
+    await _getNextMessage(nextMessageId).loadValue(ChatMsg.methodMessageGetTimestamp);
   }
 
-  Contact _getContact() {
-    return _contactRepository.get(_contactId);
+  Future<void> _setupContact() async {
+    ChatMsg message = _getMessage();
+    _contactId = await message.getFromId();
+    _contactRepository.putIfAbsent(id: _contactId);
+    await _getContact().loadValues(keys: [
+      Contact.methodContactGetName,
+      Contact.methodContactGetAddress,
+      Contact.methodContactGetColor,
+    ]);
   }
 
   ChatMsg _getMessage() {
     return _messageListRepository.get(_messageId);
   }
 
-  ChatMsg _getNextMessage() {
-    return _messageListRepository.get(_nextMessageId);
+  ChatMsg _getNextMessage(int nextMessageId) {
+    return _messageListRepository.get(nextMessageId);
   }
```

Comment:
What about:
```
ChatMsg _getMessage({@required int messageId}) {
   return _messageListRepository.get(messageId);
}

ChatMsg _getNextMessage({@required int messageId}) {
   return _getMessage(messageId: messageId);
}
```
So we can use `id` agnostic functions and only one have access to the repository.

Reply:
I like that :+1: and simplified it a bit more. If we push the id with a named parameter in anyway, we can just use one method I would say: https://github.com/open-xchange/ox-coi/pull/292/commits/c98e718dc7057a69e0627e8493904c1218a37899
I also added auto formatting (as soon as we have a two parameters, all should have a comma at the end and a new line should start behind the comma).

Reply:
What's your opinion @phranck ?

Reply:
Great!